test/test_divide_conquer.jl

```diff
@@ -0,0 +1,227 @@
+using DecisionTree: Leaf, Node
+
+@testset verbose = true "Search procedure divide and conquer" begin
+	@testset verbose = true "divide, stopping criteria" begin
+		spec::Vector{IOExample} = [
+			IOExample(Dict(:var1 => 1), 1),
+			IOExample(Dict(:var1 => 2), 2),
+			IOExample(Dict(:var1 => 3), 3),
+		]
+		problem = Problem(spec)
+		subproblems = divide(problem)
+		@test length(subproblems) == 3
+
+		# Stopping criteria: stop search once we have a solution to each subproblem
+		solutions = [RuleNode(3), RuleNode(4)]
+		problems_to_solutions::Dict{Problem, Vector{Int}} = Dict(p => [] for p in subproblems)
+
+		push!(problems_to_solutions[subproblems[1]], 1)
+		@test all(!isempty, values(problems_to_solutions)) == false
+
+		push!(problems_to_solutions[subproblems[1]], 2)
+		push!(problems_to_solutions[subproblems[2]], 1)
+		@test all(!isempty, values(problems_to_solutions)) == false
+
+		push!(problems_to_solutions[subproblems[3]], 1)
+		@test all(!isempty, values(problems_to_solutions)) == true
+
+	end
+
+	@testset verbose = true "decide" begin
+		grammar = @csgrammar begin
+			Number = |(1:2)
+			Number = x
+			Number = Number + Number
+			Number = Number * Number
+		end
+		symboltable = grammar2symboltable(grammar)
+		problem1 = Problem([IOExample(Dict(:x => 1), 3)])
+		problem2 = Problem([IOExample(Dict(:x => 1), 4)])
+		program = RuleNode(4, [RuleNode(3), RuleNode(2)])
+		expr = rulenode2expr(program, grammar)
+		@test decide(problem1, expr, symboltable) == true
+		@test decide(problem2, expr, symboltable) == false
+	end
+
+	@testset verbose = true "conquer" begin
```

Comment:
Maybe I'm missing it, but you never call the conquer function directly here, right? It looks like the tests dissect the internal logic of the function, but if the function definition is changed, these tests might not catch it. Can you add a test to check it directly?

Reply:
True, not in `test_divide_conquer.jl`. The function is called in `test_divide_conquer_example.jl` within the function call `divide_and_conquer()`.  But I added a test anyway.